### PR TITLE
PE-5423: Ready For QA: Architecture Refactor and Sync GraphQL Improvements

### PR DESCRIPTION
**Pull request by thiagocarvalhodev**
--- Releases ---
Android release: https://appdistribution.firebase.google.com/testerapps/1:305132849030:android:6cf0cd5ec064fad3ffce07/releases/298a1hvhsp4og

### CODE REVIEW

File: lib/sync/domain/cubit/sync_cubit.dart [hunk 1]
```diff
@@ -51,20 +30,15 @@ const kRequiredTxConfirmationPendingThreshold = 60 * 8;
 const kArConnectSyncTimerDuration = 2;
 const kBlockHeightLookBack = 240;
```
Comment (karlprieb) [inline lib/sync/domain/cubit/sync_cubit.dart:31]: Get these values from `constants.dart`

File: lib/sync/domain/cubit/sync_cubit.dart [hunk 2]
```diff
@@ -239,153 +211,37 @@ class SyncCubit extends Cubit<SyncState> {
...
-            driveDao: _driveDao,
-            arweave: _arweave,
-            txsIdsToSkip: confirmedFileTxIds,
-          ),
-        ],
-      );
+          })) {
+        _syncProgress = syncProgress;
+        syncProgressController.add(_syncProgress);
+      }
```
Comment (karlprieb) [inline lib/sync/domain/cubit/sync_cubit.dart:243]: create new line

**Review by karlprieb — COMMENTED** [review]
change `_lastSync` name to `_lastSyncTime` or something to make it easier to understand

File: lib/sync/domain/repositories/sync_repository.dart [hunk 3]
```diff
@@ -0,0 +1,1271 @@
...
+    final drives = await _driveDao.allDrives().map((d) => d).get();
+
+    if (drives.isEmpty) {
+      yield SyncProgress.emptySyncCompleted();
+      _lastSync = DateTime.now();
+    }
+
+    SyncProgress syncProgress = SyncProgress.initial();
+
+    syncProgress = syncProgress.copyWith(drivesCount: drives.length);
```
Comment (karlprieb) [inline lib/sync/domain/repositories/sync_repository.dart:146]:
```suggestion
    SyncProgress syncProgress = SyncProgress.initial()..copyWith(drivesCount: drives.length);
```

File: lib/sync/domain/repositories/sync_repository.dart [hunk 4]
```diff
@@ -0,0 +1,1271 @@
...
+          )
+          .getSingleOrNull();
+
+      final folderExists = folder != null;
+
+      if (folderExists) {
+        continue;
+      }
+
+      // Add to database
```
Comment (karlprieb) [inline lib/sync/domain/repositories/sync_repository.dart:299]: ?

File: lib/sync/domain/repositories/sync_repository.dart [hunk 5]
```diff
@@ -0,0 +1,1271 @@
...
+    /// Remove all confirmed transactions from the pending map
+    /// and update the status of the remaining ones
+
+    logger.i(
+      'Skipping status update for ${txsIdsToSkip.length} transactions that were captured in snapshots',
+    );
+
+    for (final txId in txsIdsToSkip) {
+      pendingTxMap.remove(txId);
+    }
```
Comment (karlprieb) [inline lib/sync/domain/repositories/sync_repository.dart:416]: Do we remove these confirmed txs from the database at some point or we keep them there and remove from the map like this?

File: lib/sync/domain/repositories/sync_repository.dart [hunk 6]
```diff
@@ -0,0 +1,1271 @@
...
+                transactionDateCreated: Value(transactionDateCreated),
+                id: Value(txId),
+                status: Value(txStatus),
+              ),
+            );
+          }
+        }
+      });
+
+      await Future.delayed(const Duration(milliseconds: 200));
```
Comment (karlprieb) [inline lib/sync/domain/repositories/sync_repository.dart:493]: why we wait 200ms here?

File: lib/sync/domain/repositories/sync_repository.dart [hunk 7]
```diff
@@ -0,0 +1,1271 @@
...
+    return rev.first.dateCreated;
+  }
+
+  bool _isOverThePendingTime(DateTime? transactionCreatedDate) {
+    // If don't have the date information we cannot assume that is over the pending time
+    if (transactionCreatedDate == null) {
+      return false;
+    }
+
+    return DateTime.now().isAfter(transactionCreatedDate.add(pendingWaitTime));
```
Comment (karlprieb) [inline lib/sync/domain/repositories/sync_repository.dart:533]: from where do we get `pendingWaitTime`

File: test/services/arweave/arweave_service_test.dart [hunk 8]
```diff
@@ -1,88 +1,61 @@
...
-
-      test('returns null for non-existant file id', () async {
-        final fileEntities = await arweave.getAllFileEntitiesWithId(
-          unknownFileId,
-        );
-        expect(fileEntities, equals(null));
-      });
-    });
-  });
+  // TODO: Fix this test after implementing the fakeNodesStream emiting DriveEntityHistoryTransactionModel
```
Comment (karlprieb) [inline test/services/arweave/arweave_service_test.dart:3]: update tests

File: test/utils/drive_history_composite_test.dart [hunk 9]
```diff
@@ -1,134 +1,121 @@
...
-import 'snapshot_test_helpers.dart';
-
 void main() {
-  group('DriveHistoryComposite class', () {
-    final arweave = MockArweaveService();
-    final List<Range> mockSubRanges = [
-      Range(start: 11, end: 25),
-      Range(start: 51, end: 98),
-    ];
+  // TODO: Fix this test after implementing the fakeNodesStream emiting DriveEntityHistoryTransactionModel
```
Comment (karlprieb) [inline test/utils/drive_history_composite_test.dart:2]: update tests

File: test/utils/gql_drive_history_test.dart [hunk 10]
```diff
@@ -1,89 +1,79 @@
...
-import 'package:flutter_test/flutter_test.dart';
-import 'package:mocktail/mocktail.dart';
-
-import '../test_utils/utils.dart';
-import 'snapshot_test_helpers.dart';
-
 void main() {
-  group('GQLDriveHistory class', () {
-    final arweave = MockArweaveService();
+  // TODO: Fix this test after implementing the fakeNodesStream emiting DriveEntityHistoryTransactionModel
```
Comment (karlprieb) [inline test/utils/gql_drive_history_test.dart:2]: update tests

File: test/utils/snapshot_item_to_be_created_test.dart [hunk 11]
```diff
@@ -1,160 +1,151 @@
...
-          driveId: 'DRIVE_ID',
-          blockStart: 0,
-          blockEnd: 10,
-          subRanges: HeightRange(rangeSegments: [Range(start: 0, end: 10)]),
-          source: const Stream.empty(),
-          jsonMetadataOfTxId: (txId) async => Uint8List.fromList(
-            utf8.encode('{"name":"$txId"}'),
-          ),
-        );
+  // TODO:
```
Comment (karlprieb) [inline test/utils/snapshot_item_to_be_created_test.dart:2]: update tests

**Review by karlprieb — APPROVED** [review]

File: lib/sync/domain/repositories/sync_repository.dart (hunk 6 above)
Comment (thiagocarvalhodev) [reply lib/sync/domain/repositories/sync_repository.dart:493]: Was a delay we added between insetion of each page. Not sure why

**Review by karlprieb — APPROVED** [review]

**Pull request merged**